rhino/src/main/java/org/mozilla/javascript/NativeSet.java

```diff
@@ -152,7 +211,12 @@ private Object js_delete(Object arg) {
     }
 
     private Object js_has(Object arg) {
-        return entries.has(arg);
+        // Special handling of "negative zero" from the spec.
```

Comment:
Maybe something like this is a bit more clear

    // Special handling of "negative zero" from the spec.
    if ((arg instanceof Number) && ((Number) arg).doubleValue() == ScriptRuntime.negativeZero) {
        return entries.has(ScriptRuntime.zeroOb)j;
    }
    return entries.has(arg);

But this is only my personal taste